Source/FieldSolver/SpectralSolver/SpectralAlgorithms/PsatdAlgorithm.cpp

```diff
@@ -672,6 +770,76 @@ void PsatdAlgorithm::InitializeSpectralCoefficientsAveraging (
     }
 }
 
+void PsatdAlgorithm::InitializeSpectralCoefficientsAvgLin (
+    const SpectralKSpace& spectral_kspace,
+    const amrex::DistributionMapping& dm,
+    const amrex::Real dt)
+{
+    const BoxArray& ba = spectral_kspace.spectralspace_ba;
+
+    // Loop over boxes and allocate the corresponding coefficients for each box
+    for (MFIter mfi(ba, dm); mfi.isValid(); ++mfi)
+    {
+        const Box& bx = ba[mfi];
+
+        // Extract pointers for the k vectors
+        const Real* kx_s = modified_kx_vec[mfi].dataPtr();
+#if (AMREX_SPACEDIM==3)
+        const Real* ky_s = modified_ky_vec[mfi].dataPtr();
+#endif
+        const Real* kz_s = modified_kz_vec[mfi].dataPtr();
+
+        Array4<Real> C = C_coef[mfi].array();
+        Array4<Real> S_ck = S_ck_coef[mfi].array();
+
+        Array4<Complex> X5 = X5_coef[mfi].array();
+        Array4<Complex> X6 = X6_coef[mfi].array();
```

Comment:
You can probably make those const as above:
```suggestion
        Array4<Real const> C = C_coef[mfi].array();
        Array4<Real const> S_ck = S_ck_coef[mfi].array();

        Array4<Complex const> X5 = X5_coef[mfi].array();
        Array4<Complex const> X6 = X6_coef[mfi].array();
```

Reply:
Done in https://github.com/ECP-WarpX/WarpX/pull/1828/commits/971665564e46b1f24df0308c886ccbd8018fab90, except for `X5` and `X6`, which are assigned after these declarations.